spec/controllers/conference_registration_controller_spec.rb

```diff
@@ -0,0 +1,284 @@
+require 'spec_helper'
+
+describe ConferenceRegistrationsController, type: :controller do
+  let!(:first_user) { create(:user) }
+  let(:conference) { create(:conference) }
+  let(:conference_with_open_registration) { create(:conference) }
+  let!(:open_registration_period) { create(:registration_period, conference: conference_with_open_registration, start_date: Date.current - 6.days) }
+  let(:user) { create(:user) }
+
+  context 'user is not signed in' do
+    describe 'GET #new' do
+      context 'ichain is enabled' do
```

Comment:
yes! For [following else](https://github.com/openSUSE/osem/blob/master/app/controllers/conference_registrations_controller.rb#L13-L16):
```Ruby
 # Redirect to registration edit when user is already registered
    if @conference.user_registered?(current_user)
      redirect_to edit_conference_conference_registrations_path(@conference.short_title)
      return
    # ichain does not allow us to create users during registration
    elsif CONFIG['authentication']['ichain']['enabled'] && !current_user
      redirect_to root_path, alert: 'You need to sign in or sign up before continuing.'
      return
    end
```